deploy/install.sh

```diff
@@ -176,18 +175,59 @@ EOF
 }
 
 checkForExistingCluster() {
-  response="$(curl -sS -X GET "${API_URL}/config/v1/kubernetes/credentials" \
-    -H "accept: application/json; charset=utf-8" \
-    -H "Authorization: Api-Token ${API_TOKEN}" \
-    -H "Content-Type: application/json; charset=utf-8")"
-
+  response=$(apiRequest "GET" "/config/v1/kubernetes/credentials" "")
   if echo "$response" | grep -FEq "\"name\":\"${CLUSTER_NAME}\""; then
     echo "Error: Cluster name already exists!"
     exit 1
   fi
 }
 
+checkTokenScopes() {
+  jsonAPI="{\"token\": \"${API_TOKEN}\"}"
+  jsonPaaS="{\"token\": \"${PAAS_TOKEN}\"}"
+
+  responseAPI=$(apiRequest "POST" "/v1/tokens/lookup" "${jsonAPI}")
+
+  if echo "$responseAPI" | grep -q "Authentication failed"; then
+    echo "Error: API token authentication failed!"
+    exit 1
+  fi
+
+  if ! echo "$responseAPI" | grep -q "WriteConfig"; then
+    echo "Error: API token does not have config write permission!"
+    exit 1
+  fi
+
+  if ! echo "$responseAPI" | grep -q "ReadConfig"; then
+    echo "Error: API token does not have config read permission!"
+    exit 1
+  fi
+
+  responsePaaS=$(apiRequest "POST" "/v1/tokens/lookup" "${jsonPaaS}")
+
+  if echo "$responsePaaS" | grep -q "Token does not exist"; then
```

Comment:
Technically, you could check for the `InstallerDownload` permission for PaaS tokens.

Reply:
Since the PaaSToken can only have InstallerDownload permission checking for it's existence should be sufficient here

Reply:
Except when the customer uses an API token as PaaS token, which is not that uncommon.